File: code/python/chainofresponsibility/LevelTwoSupport.py
from SupportLevel import SupportLevel


class LevelTwoSupport(SupportLevel):
    def __init__(self):
        self.next = None

    def set_next(self, next):
        print(f"Setting next level to {type(next).__name__} Support")
        self.next = next

    def handle_request(self, ticket):
        print(f"Level Two Support handling ticket: {ticket.get_description()}")
        print("Ticket resolved.")

File: code/typescript/src/chainofresponsibility/LevelTwoSupport.ts
import { SupportLevel } from "./SupportLevel";
import { SupportTicket } from "./SupportTicket";

export class LevelTwoSupport implements SupportLevel {
    private next?: SupportLevel;

    setNext(next: SupportLevel): void {
        this.next = next;
        console.log(`Setting next level to ${next.constructor.name} Support`);
    }

    handleRequest(ticket: SupportTicket): void {
        console.log(`Level Two Support handling ticket: ${ticket.description}`);
        console.log("Ticket resolved.");
    }
}
